reproducible/README.md

```diff
@@ -0,0 +1,343 @@
+```
+will perform both instantiate and realise. The result is a build output located in the nix-store.
+```
+/nix/store/syibqm72433pzxd46hyw3xbr5swq286j-test-environment
+```
+which contains:
+```
+$ ls /nix/store/syibqm72433pzxd46hyw3xbr5swq286j-test-environment
+qsgajw2yhrf33qgibsvd9y8zaw64w7j7-helloWorld
+```
+which in turn contains the helloWorld script.
+
+For more involved build project the build actions are clearly more complex, but the flow is the same.
+
+### The Nix Repo
+
+To build, a clone of the nix packages repo must be present. There is a default clone installed when nix 
+is installed. 
+
+The nix packages repo serves two purposes, somewhat at odds with each other.
+
+- The package repository for a linux distro "NixOS" which will eventully be a standalone distro. In that 
+  case packages are chosen for release branches with an eye to mutual compatibility and working well within
+  that defined universe.  
+
+- The package repository is also used for producing packages for other linux distros (ubuntu, debian,
+  redhat, fedora, arch, gentoo) with easily auditable dependencies. In this case, it is necessary to
+  consider specific compatible versions of packages.  In particular, glibc must be an equal or older
+  revision than the glibc of the platform or the program will not dynamically link.  This is by design.
+  Ubuntu 18.04 is limited to glibc 2.27. The current lts release 20.04 usues glibc 2.31 and fedora 33 
+  uses libc-2.32.  The nixpkgs branch release-20.09 is based on glibc 2.32.  The somewhat older branch
+  20.03 uses libc-2.30 so can work on the current common distros, including the current rhel.  
+
+  While it is not difficult to change the version of glibc in a custom repo branch, since glibc is linked 
+  by every package, to do so would require a custom repo build. In that case, it would require a custom 
+  cache to provide prebuilding, or everything would require everything be built from source..
+
+### Caching and Hydra
+
+The official branches in nixpkgs are prebuilt and available at cache.nixos.org. Caching is keyed to the build outputs
+sha digest, so custom environments will be able to use those prebuilts so long as no changes have been made. 
+If any changes are made to that package, any package related to the changed package will need to be custom
+built and cached in nix store.
+
+
+## A Nix build system for OpenEnclave SDK
+
+We propose to add an optional nix based build environment for OpenEnclave SDK using docker containers.
+
+This would consist of :
+- Dockerfile.nix and build-oe-nix-build.sh for building the oe nix docker image locally.
+- build_openenclave.sh for building the oe deb package via a nix environment.
+- Dockerfile.test and test-oe-nix-build.sh for verifying that package is functional.
+
+The process builds a reproducible .deb package of openenclave-sdk. 
+
+This process can be integrated with ADO or CPDX CI, but that is not part of this submission. We do not currently
```

Comment:
CDPx is a Microsoft thing, a better phrasing would be that it can be integrated with all CI build engines that support containers.

Reply:
Done.